### ISSUE TITLE
Unknown provider: $swipeProvider <- $swipe <- rnCarouselDirective

### ISSUE THREAD

**Issue by RohitRane123** [revolunet/angular-carousel#342]
I followed the steps you mentioned in your guidelines inorder to include your carousel in my app 

This is script code 
var myApp = angular.module('MyApp', ['angular-carousel']);

myApp.controller('MyController',function($scope){

});

And this is my html code 

<!DOCTYPE html>
<head>
	<link href="angular-carousel.css" rel="stylesheet" type="text/css" />
</head>
<body>
	<script src="https://cdnjs.cloudflare.com/ajax/libs/angular.js/1.4.2/angular.js"></script>
<script src="angular-touch.js"></script>
<script src="angular-carousel.js"></script>
<script src="myApp.js"></script>
<div ng-app="MyApp">
<div ng-controller="MyController">
<ul rn-carousel class="image">
  <li>slide #1</li>
  <li>slide #2</li>
  <li>slide #3</li>
</ul>
</div>
</div>
</body>
</html>

Yet I am still getting the error 

[$injector:unpr] Unknown provider: $swipeProvider <- $swipe <- rnCarouselDirective
http://errors.angularjs.org/1.4.2/$injector/unpr?p0=%24swipeProvider%20%3C-%20%24swipe%20%3C-%20rnCarouselDirective
    at REGEX_STRING_REGEXP (angular.js:68)
    at angular.js:4264
    at Object.getService [as get] (angular.js:4411)
    at angular.js:4269
    at getService (angular.js:4411)
    at Object.invoke (angular.js:4443)
    at angular.js:6943
    at forEach (angular.js:336)
    at Object.<anonymous> (angular.js:6941)
    at Object.invoke (angular.js:4452)

What am I doing wrong in my implementation.

**Comment by revolunet** [comment]
did you add angular-touch.js as mentioned in the [README](https://github.com/revolunet/angular-carousel/blob/master/README.md) ?

**Comment by RohitRane123** [comment]
Yes.. I fixed the error  I had to rewrite my module definition as : 
angular.module('MyApp', ['angular-carousel'])
.controller('MyController',function($scope){

});

**Comment by RohitRane123** [comment]
I thought the issue was resolved but it's its not.

It just that in the html tag when I changed the directive name from rn-carousel to rnCarousel the error went away but carousel was not generated. So please re-open the issue.

**Comment by revolunet** [comment]
ok, please provide a plunker with reproduction possible

**Comment by RohitRane123** [comment]
Actually I needed it for a project to be deployed in crunch time. So, I switched to another jQuery based carousel. But, I'll try and reproduce in plunker and share here ASAP.

**Comment by arjundas** [comment]
i also have this problem.

**Comment by ramoncarreras** [comment]
I have this problem too. (AngularJS 1.5)

**Comment by PeterStegnar** [comment]
Do like this (Webpack solution):

import angularTouch from 'angular-touch';
import 'angular-carousel';

And then

const carousel = 'angular-carousel';
const ngTouch = 'angular-touch';

and use those variables in your modules dependencies.

[ ..., angularTouch, carousel  ],

Hope it helps!

**Comment by nlaycock** [comment]
did anyone find a solution that does not use webpack? I am on ng 1.5.11

**Comment by Reicr** [comment]
We had the same problem in our project and found out that angular-carousel was loading angular-touch version 1.6.x which was most likely not compatible with our angular version 1.5.11.

To avoid loading an incompatible version we explicitly specified the angular-touch version we want in our bower.json which resolved the problem for us.

**Comment by degroundshaker** [comment]
I solved this.

Issue was **angular-touch.js** version incompatibility.

I was using **Ver. 1.6.4** before but then downgraded it to **1.5.8** & its working fine. 

One more thing to consider is to avoid using **angular.js** twice. 

In my ionic app, it is included by default so I had to remove one which I had already added under my index.php

Also dont forget to style your .image class by setting its display to block & proper height width attributes.
